hknweb/events/views.py

```diff
@@ -2,17 +2,35 @@
 from django.http import HttpResponse
 from django.template import loader
 
-from .models import Event
+from .models import Event, Rsvp
 
 def index(request):
-    events = Event.objects.order_by('-created_at')[:10]
-
+    events = Event.objects.order_by('-start_time')[:10]
     context = {
         'events': events,
     }
     return render(request, 'events/index.html', context)
 
 
+def show_details(request, id):
+    event = Event.objects.get(pk=id)
+    context = {
+        'event': event,
+    }
+    return render(request, 'events/show_details.html', context)
+
+
+def rsvp(request, id):
+    # If authenticated and event is under RSVP cap
+    event = Event.objects.get(pk=id)
+    print("rsvps:", event.rsvps)
+    if request.user.is_authenticated and event.rsvps < event.rsvp_limit:
+        event.rsvps += 1
+        event.save()
```

Comment:
This may not be an issue, but this may cause a [race condition](https://en.wikipedia.org/wiki/Race_condition): two simultaneous rsvps which both check the `rsvp_limit` may both succeed, but then push the `event.rsvps` beyond `rsvp_limit`.

I'm not sure how to solve this (might be a non-issue, if the rate of responses is low enough), but the technical solution would be either atomic database updates (making sure limit read + rsvp increment cannot be done simultaneously) or transactional updates, below:
https://docs.djangoproject.com/en/2.1/topics/db/transactions/#controlling-transactions-explicitly